vim_menutrans_helper.py

```diff
@@ -0,0 +1,196 @@
+             "xme", "xmenu", "xnoreme", "xnoremenu", "sme", "smenu",
+             "snoreme", "snoremenu", "ime", "imenu", "inoreme", "inoremenu",
+             "cme", "cmenu", "cnoreme", "cnoremenu", "tlm", "tlmenu", "tln",
+             "tlnoremenu", "E330", "E327", "E331", "E336", "E333", "E328", "E329",
+             "E337", "E792", "tunmenu", "tu", "popup", "popu"}
+toolbar_item = {"tmenu", "tm"}
+
+# unescape double quotes
+def unescape_double_quotes(origin):
+    origin = origin.replace('\\"', '"')
+    origin = origin.replace("\\'", "'")
+    origin = origin.replace('\\t', '\t')
+    origin = origin.replace('\\\\', '\\')
+    return origin
+
+# unescape single quotes
+def unescape_single_quotes(origin):
+    origin = origin.replace("''", "'")
+    return origin
+
+# get the diffence between untranslated file and translated file
+def makeUnTranslatedDic(untranslated_file, untranslated_dict):
+    # deal with the file need to be translated
+    with open(untranslated_file, encoding="latin1") as f1:
+        for line_number, line in enumerate(f1):
+            tobe_translated_words = []
+
+            line = line.strip()
+            line = line.rstrip()
+
+            temp_word_list = []
+            pre_end = -1
+            top_string = line.split()
+            if len(top_string) > 0 and (top_string[0] == "exe" or top_string[0] == "exec" or top_string[0] == "execute"):
+                # extract content inside quotes from line
+                for ite in re.finditer(r"'(?:''|[^'])*'|(?:\"(?:(?:\\.)|[^\"])*\")", line):
+                    mid_part = line[pre_end + 1: ite.start() - 1].strip()
+                    if not ((len(mid_part) == 1 and mid_part[0] == ".") or len(mid_part) == 0 or pre_end == -1):
+                        temp_word_list.append("5")
+                    pre_end = ite.end()
+                    temp_word_list.append(line[ite.start() + 1: ite.end() - 1])
+
+                temp_line = ""
+                for word in temp_word_list:
+
+                    if len(word) == 0:
+                        continue
+                    if word[0] == "'":
+                        word = unescape_single_quotes(word)
+                    else:
+                        word = unescape_double_quotes(word)
+                    temp_line += word
+                line = temp_line
+
+            word_list = []
+            # split line string according to blank without backslash or tab ahead
+            for ite in re.finditer(r"(?:(?:\\.)|[^ \t])+", line):
+                word_list.append(line[ite.start(): ite.end()])
+
+            if len(word_list) == 0 or word_list[0] == '"':
+                continue
+            index = 0
+            # menu_item with "disable" or enable followed close behind should be ignored
+            if word_list[0] in menu_item and len(word_list) > 1 and (word_list[1] == "disable" or word_list[1] == "enable"):
+                continue
+
+            if word_list[0] == 'sil!':
```

Comment:
`sil` can also be written as `silent`. And `!` at the end is optional. Maybe you should define a set here to cover all the possibilities.